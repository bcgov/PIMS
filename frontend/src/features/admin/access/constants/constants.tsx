import { ColumnWithProps } from 'components/Table';
import { AccessStatusDisplayMapper } from 'constants/accessStatus';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

import { RowActions } from '../components/RowActions';
import { IAccessRequestModel } from '../interfaces';

export const columnDefinitions: ColumnWithProps<IAccessRequestModel>[] = [
  {
    Header: 'IDIR/BCeID',
    accessor: 'username',
    align: 'left',
    Cell: (props: CellProps<IAccessRequestModel>) => {
      return (
        <Link to={`/admin/user/${props.row.original.userId}`}>{props.row.original.username}</Link>
      );
    },
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Last name',
    accessor: 'lastName',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Status',
    accessor: 'status',
    align: 'left',
    clickable: true,
    width: 100,
    Cell: (props: CellProps<IAccessRequestModel>) =>
      AccessStatusDisplayMapper[props.row.original.status],
  },
  {
    Header: 'Agency',
    accessor: 'agency',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Role',
    accessor: 'role',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: ' ',
    Cell: RowActions,
    width: 75,
  },
];
