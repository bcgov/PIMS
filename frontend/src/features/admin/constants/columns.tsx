import { ColumnWithProps } from 'components/Table';
import { IAgency, IAgencyRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IAgencyRecord>[] = [
  {
    Header: 'Agency name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Agency',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IAgency>) => {
      return (
        <Link to={`/admin/agency/${props.row.original.parentId}`}>{props.row.original.parent}</Link>
      );
    },
  },
];

export const adminAreasColumnDefinistions: ColumnWithProps<any>[] = [
  {
    Header: 'Area name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
];
