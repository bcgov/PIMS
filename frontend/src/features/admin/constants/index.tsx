import * as React from 'react';
import { ColumnWithProps } from 'components/Table';
import { CellProps } from 'react-table';
import { Link } from 'react-router-dom';
import { IAgency, IAgencyRecord } from 'interfaces';

export const columnDefinitions: ColumnWithProps<IAgencyRecord>[] = [
  {
    Header: 'Agency name',
    accessor: 'name',
    align: 'left',
    sortable: true,
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Parent Agency',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IAgency>) => {
      return (
        <Link to={`/admin/agency/${props.row.original.parentId}`}>{props.row.original.parent}</Link>
      );
    },
    sortable: true,
  },
];
