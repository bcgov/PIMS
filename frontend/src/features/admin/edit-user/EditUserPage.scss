.userInfo {
  width: 40%;
  vertical-align: middle;
  text-align: left;
  display: table;
}

.notes {
  padding-bottom: 30px;
}

.is-disabled {
  display: flex;
  flex-direction: row;
  align-items: center;
  :first-child {
    margin-right: 10px;
    margin-bottom: 0px;
  }
}

.cancelSave {
  padding-top: 15px;
  padding-bottom: 10px;
}

.navBar {
  font-weight: 700;
  text-align: left;
}

.required {
  color: red;
  font-weight: 700;
}

.user-edit-form-container {
  justify-content: center;
}
