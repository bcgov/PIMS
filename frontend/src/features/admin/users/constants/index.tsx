import { ColumnWithProps } from 'components/Table';
import * as React from 'react';
import { FaCheck, FaTimes } from 'react-icons/fa';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

import { RowActions } from '../components/RowActions';
import { IUserRecord } from '../interfaces/IUserRecord';

export const columnDefinitions: ColumnWithProps<IUserRecord>[] = [
  {
    Header: 'IDIR/BCeID',
    accessor: 'username',
    align: 'left',
    Cell: (props: CellProps<IUserRecord>) => {
      return <Link to={`/admin/user/${props.row.original.id}`}>{props.row.original.username}</Link>;
    },
    sortable: true,
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Last name',
    accessor: 'lastName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Active',
    accessor: 'isDisabled',
    align: 'left',
    clickable: true,
    sortable: true,
    width: 100,
    Cell: (props: CellProps<IUserRecord>) =>
      props.row.original.isDisabled ? <FaTimes /> : <FaCheck />,
  },
  {
    Header: 'Agency',
    accessor: 'agency',
    align: 'left',
    sortable: true,
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Roles',
    accessor: 'roles',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Last Login',
    accessor: 'lastLogin',
    align: 'left',
    clickable: true,
    minWidth: 100,
  },
  {
    Header: 'Created On',
    accessor: 'createdOn',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 100,
  },
  {
    Header: ' ',
    Cell: RowActions,
    width: 75,
  },
];
