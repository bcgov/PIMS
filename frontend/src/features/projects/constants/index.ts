export * from './AgencyResponses';
export * from './DisposalWorkflows';
export * from './DisposeWorkflowStatus';
export * from './ProjectWorkflowComponents';
export * from './ReviewWorkflowStatus';
export * from './SPPApprovalTabs';
