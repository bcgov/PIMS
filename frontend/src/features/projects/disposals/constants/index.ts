export * from './defaultFilter';
export * from './defaultProjectForm';
