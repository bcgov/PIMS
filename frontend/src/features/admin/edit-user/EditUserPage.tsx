import React, { useEffect } from 'react';
import { Navbar, Container, Row, Col, ButtonToolbar, Button } from 'react-bootstrap';
import { Form, Input, Select, SelectOption } from '../../../components/common/form';
import { fetchUserDetail, getUpdateUserAction } from 'actionCreators/usersActionCreator';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from 'reducers/rootReducer';
import { IUserDetails } from 'interfaces';
import { Formik, Field } from 'formik';
import { UserUpdateSchema } from 'utils/YupSchema';
import { IUserDetailParams } from 'constants/API';
import * as API from 'constants/API';
import './EditUserPage.scss';
import { Label } from 'components/common/Label';
import { useHistory } from 'react-router-dom';
import TooltipIcon from 'components/common/TooltipIcon';
import { formatApiDateTime } from 'utils';
import useCodeLookups from 'hooks/useLookupCodes';
import { ILookupCode } from 'actions/lookupActions';

interface IEditUserPageProps extends IUserDetailParams {
  match?: any;
}

const EditUserPage = (props: IEditUserPageProps) => {
  const userId = props?.match?.params?.id || props.id;
  const history = useHistory();
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchUserDetail({ id: userId }));
  }, [dispatch, userId]);

  const { getByType } = useCodeLookups();
  const agencies = getByType(API.AGENCY_CODE_SET_NAME);
  const roles = getByType(API.ROLE_CODE_SET_NAME);

  const user = useSelector<RootState, IUserDetails>(state => state.GET_USER_DETAIL as IUserDetails);
  const mapLookupCode = (code: ILookupCode): SelectOption => ({
    label: code.name,
    value: code.id.toString(),
    selected: !!user.roles.find(x => x.id === code.id.toString()),
  });

  const selectAgencies = agencies.map(c => mapLookupCode(c));
  const selectRoles = roles.map(c => mapLookupCode(c));

  // Arrays below are used to add the role/agency from the dropdown later in code
  let agenciesToUpdate: any[];
  let rolesToUpdate: any[];

  const checkAgencies = (
    <Select
      label="Agency"
      field="agency"
      data-testid="agency"
      required={true}
      options={selectAgencies}
      placeholder={user?.agencies?.length > 0 ? undefined : 'Please Select'}
    />
  );

  const checkRoles = (
    <Select
      label="Roles"
      field="roles"
      data-testid="role"
      multiple={true}
      required={true}
      options={selectRoles}
      placeholder={user?.roles?.length > 0 ? undefined : 'Please Select'}
    />
  );

  const goBack = () => {
    history.goBack();
  };

  const initialValues = {
    username: user.username,
    firstName: user.firstName,
    lastName: user.lastName,
    email: user.email,
    displayName: user.displayName,
    isDisabled: !!user.isDisabled,
    rowVersion: user.rowVersion,
    emailVerified: false,
    agencies: user.agencies,
    roles: user.roles.map(x => x.id),
    note: user.note,
    agency: user.agencies && user.agencies.length !== 0 ? user.agencies[0].id : '',
    role: user.roles && user.roles.length !== 0 ? user.roles[0].id : '',
    position: user.position,
    lastLogin: formatApiDateTime(user.lastLogin),
  };

  return (
    <div>
      <Navbar className="navBar" expand="sm" variant="light" bg="light">
        <Navbar.Brand href="#">User Information</Navbar.Brand>
      </Navbar>
      <Container fluid={true}>
        <Row className="user-edit-form-container">
          <Formik
            enableReinitialize
            initialValues={initialValues}
            validationSchema={UserUpdateSchema}
            onSubmit={(values, { setSubmitting }) => {
              if (values.agency !== '') {
                agenciesToUpdate = [{ id: Number(values.agency) }];
              } else {
                agenciesToUpdate = user.agencies;
              }

              if (values.roles) {
                rolesToUpdate = values.roles.map(r => ({ id: r }));
              } else {
                rolesToUpdate = user.roles;
              }
              dispatch(
                getUpdateUserAction(
                  { id: userId },
                  {
                    id: user.id,
                    username: user.username,
                    displayName: values.displayName,
                    firstName: values.firstName,
                    lastName: values.lastName,
                    email: values.email,
                    isDisabled: values.isDisabled,
                    rowVersion: values.rowVersion,
                    emailVerified: values.emailVerified,
                    agencies: agenciesToUpdate,
                    roles: rolesToUpdate,
                    position: values.position,
                    note: values.note,
                  },
                ),
              );
              setSubmitting(false);
            }}
          >
            {props => (
              <Form className="userInfo">
                <Label>IDIR/BCeID</Label>
                <Input
                  data-testid="username"
                  field="username"
                  value={props.values.username}
                  readOnly={true}
                  type="text"
                />
                <Label>Last Login</Label>
                <Input
                  data-testid="lastLogin"
                  field="lastLogin"
                  value={props.values.lastLogin}
                  readOnly={true}
                  type="text"
                />

                <Row>
                  <Col>
                    <Label>First Name</Label>
                    <Input
                      data-testid="firstName"
                      field="firstName"
                      placeholder={props.values.firstName}
                      type="text"
                    />
                  </Col>
                  <Col>
                    <Label>Last Name</Label>
                    <Input
                      data-testid="lastName"
                      field="lastName"
                      placeholder={props.values.lastName}
                      type="text"
                    />
                  </Col>
                </Row>

                <Label>Email</Label>
                <Input
                  data-testid="email"
                  field="email"
                  placeholder={props.values.email}
                  type="email"
                />

                {checkAgencies}

                <Label>Position</Label>
                <Input
                  field="position"
                  placeholder="e.g) Director, Real Estate and Stakeholder Engagement"
                  type="text"
                  data-testid="position"
                />

                {checkRoles}

                <Label>Note</Label>
                <Input
                  as="textarea"
                  field="note"
                  placeholder="A note about this user"
                  type="text"
                  data-testid="note"
                />

                <Form.Group className={'is-disabled'}>
                  <Form.Label>
                    Is Disabled?{' '}
                    <TooltipIcon
                      toolTipId="is-disabled-tooltip"
                      toolTip={'Toggle to change account status and click save.'}
                    />{' '}
                  </Form.Label>
                  <Field data-testid="isDisabled" type="checkbox" name="isDisabled" />
                </Form.Group>

                <Row className="justify-content-md-center">
                  <ButtonToolbar className="cancelSave">
                    <Button className="mr-5" variant="outline-dark" type="button" onClick={goBack}>
                      Cancel
                    </Button>
                    <Button className="mr-5" type="submit">
                      Save
                    </Button>
                  </ButtonToolbar>
                </Row>
              </Form>
            )}
          </Formik>
        </Row>
      </Container>
    </div>
  );
};

export default EditUserPage;
