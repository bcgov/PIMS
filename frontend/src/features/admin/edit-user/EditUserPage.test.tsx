import EditUserPage from './EditUserPage';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { ILookupCode } from 'actions/lookupActions';
import * as API from 'constants/API';
import { Provider } from 'react-redux';
import * as reducerTypes from 'constants/reducerTypes';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { render, cleanup, act } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { ToastContainer } from 'react-toastify';
import moment from 'moment-timezone';

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();

const lCodes = {
  lookupCodes: [
    { name: 'agencyVal', id: '1', isDisabled: false, type: API.AGENCY_CODE_SET_NAME },
    { name: 'disabledAgency', id: '2', isDisabled: true, type: API.AGENCY_CODE_SET_NAME },
    { name: 'roleVal', id: '1', isDisabled: false, type: API.ROLE_CODE_SET_NAME },
    { name: 'disabledRole', id: '2', isDisabled: true, type: API.ROLE_CODE_SET_NAME },
  ] as ILookupCode[],
};

const selectedUser = {
  username: 'test.user',
  firstName: 'Test',
  lastName: 'User',
  email: 'test@user.com',
  isDisabled: false,
  emailVerified: false,
  agencies: [],
  roles: [{ id: '2' }],
  rowVersion: 'AAAAAAAAB9E=',
  note: 'test note',
  lastLogin: '2020-10-14T17:45:39.7381599',
};

const store = mockStore({
  [reducerTypes.GET_USER_DETAIL]: selectedUser,
  [reducerTypes.LOOKUP_CODE]: lCodes,
});

const noDateStore = mockStore({
  [reducerTypes.GET_USER_DETAIL]: { ...selectedUser, lastLogin: null },
  [reducerTypes.LOOKUP_CODE]: lCodes,
});

const mockAxios = new MockAdapter(axios);

const testRender = () =>
  render(
    <Provider store={store}>
      <Router history={history}>
        <EditUserPage id="TEST-ID" />,
      </Router>
    </Provider>,
  );

const renderEditUserPage = () =>
  render(
    <Provider store={store}>
      <Router history={history}>
        <ToastContainer
          autoClose={5000}
          hideProgressBar
          newestOnTop={false}
          closeOnClick={false}
          rtl={false}
          pauseOnFocusLoss={false}
        />
        <EditUserPage id="TEST-ID" />,
      </Router>
    </Provider>,
  );

describe('Edit user page', () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });
  beforeEach(() => {
    mockAxios.onAny().reply(200, {});
  });
  it('EditUserPage renders', () => {
    const { container } = render(
      <Provider store={noDateStore}>
        <Router history={history}>
          <EditUserPage id="TEST-ID" />,
        </Router>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('contains role options from lookup code + please select disabled option', () => {
    const { getAllByText, getByTestId } = renderEditUserPage();
    expect(getAllByText(/Roles/i));
    expect(getAllByText(/roleVal/i));
    expect(getAllByText(/agencyVal/i));
    expect(getByTestId('isDisabled').getAttribute('value')).toEqual('false');
  });

  it('displays enabled agencies', () => {
    const { queryByText } = testRender();
    expect(queryByText('agencyVal')).toBeVisible();
  });

  it('Does not display disabled agencies', () => {
    const { queryByText } = testRender();
    expect(queryByText('disabledAgency')).toBeNull();
  });

  it('displays enabled roles', () => {
    const { queryByText } = testRender();
    expect(queryByText('roleVal')).toBeVisible();
  });

  it('Does not display disabled roles', () => {
    const { queryByText } = testRender();
    expect(queryByText('disabledRole')).toBeNull();
  });

  describe('appropriate fields are autofilled', () => {
    it('autofills  email, username, first and last name', () => {
      const { getByTestId } = renderEditUserPage();
      expect(getByTestId('email').getAttribute('value')).toEqual('test@user.com');
      expect(getByTestId('username').getAttribute('value')).toEqual('test.user');
      expect(getByTestId('firstName').getAttribute('value')).toEqual('Test');
      expect(getByTestId('lastName').getAttribute('value')).toEqual('User');
      expect(getByTestId('lastName').getAttribute('value')).toEqual('User');
    });
  });

  describe('when the user edit form is submitted', () => {
    it('displays a loading toast', async done => {
      const { getByText, findByText } = renderEditUserPage();
      const saveButton = getByText('Save');
      act(() => {
        saveButton.click();
      });
      await findByText('Updating User...');
      done();
    });

    it('displays a success toast if the request passes', async done => {
      const { getByText, findByText } = renderEditUserPage();
      const saveButton = getByText('Save');
      act(() => {
        saveButton.click();
      });
      await findByText('User updated');
      done();
    });

    it('displays a error toast if the request fails', async done => {
      const { getByText, findByText } = renderEditUserPage();
      const saveButton = getByText('Save');
      mockAxios.reset();
      mockAxios.onAny().reply(500, {});
      act(() => {
        saveButton.click();
      });
      await findByText('Failed to update User');
      done();
    });

    it('Displays the correct last login time', () => {
      const dateTime = moment
        .utc('2020-10-14T17:45:39.7381599')
        .local()
        .format('YYYY-MM-DD hh:mm a');
      const { getByTestId } = renderEditUserPage();
      expect(getByTestId('lastLogin')).toHaveValue(dateTime);
    });
  });
});
