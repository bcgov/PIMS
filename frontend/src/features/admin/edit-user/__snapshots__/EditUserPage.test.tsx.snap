// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit user page EditUserPage renders 1`] = `
"<div>
  <nav class=\\"navBar navbar navbar-expand-sm navbar-light bg-light\\"><a href=\\"#\\" class=\\"navbar-brand\\">User Information</a></nav>
  <div class=\\"container-fluid\\">
    <div class=\\"user-edit-form-container row\\">
      <form novalidate=\\"\\" class=\\"userInfo\\">
        <p class=\\"label undefined\\">IDIR/BCeID</p>
        <div class=\\"form-group\\"><input name=\\"username\\" data-testid=\\"username\\" type=\\"text\\" readonly=\\"\\" id=\\"input-username\\" class=\\"form-control\\" value=\\"test.user\\"></div>
        <p class=\\"label undefined\\">Last Login</p>
        <div class=\\"form-group\\"><input name=\\"lastLogin\\" data-testid=\\"lastLogin\\" type=\\"text\\" readonly=\\"\\" id=\\"input-lastLogin\\" class=\\"form-control\\" value=\\"\\"></div>
        <div class=\\"row\\">
          <div class=\\"col\\">
            <p class=\\"label undefined\\">First Name</p>
            <div class=\\"form-group\\"><input name=\\"firstName\\" data-testid=\\"firstName\\" placeholder=\\"Test\\" type=\\"text\\" id=\\"input-firstName\\" class=\\"form-control\\" value=\\"Test\\"></div>
          </div>
          <div class=\\"col\\">
            <p class=\\"label undefined\\">Last Name</p>
            <div class=\\"form-group\\"><input name=\\"lastName\\" data-testid=\\"lastName\\" placeholder=\\"User\\" type=\\"text\\" id=\\"input-lastName\\" class=\\"form-control\\" value=\\"User\\"></div>
          </div>
        </div>
        <p class=\\"label undefined\\">Email</p>
        <div class=\\"form-group\\"><input name=\\"email\\" data-testid=\\"email\\" placeholder=\\"test@user.com\\" type=\\"email\\" id=\\"input-email\\" class=\\"form-control\\" value=\\"test@user.com\\"></div>
        <div class=\\"required form-group\\"><label class=\\"form-label\\" for=\\"input-agency\\">Agency</label><span class=\\"required\\">*</span><select name=\\"agency\\" required=\\"\\" data-testid=\\"agency\\" id=\\"input-agency\\" class=\\"form-select form-control\\">
            <option value=\\"\\">Please Select</option>
            <option value=\\"1\\" class=\\"option\\">agencyVal</option>
          </select></div>
        <p class=\\"label undefined\\">Position</p>
        <div class=\\"form-group\\"><input name=\\"position\\" data-testid=\\"position\\" placeholder=\\"e.g) Director, Real Estate and Stakeholder Engagement\\" type=\\"text\\" id=\\"input-position\\" class=\\"form-control\\" value=\\"\\"></div>
        <div class=\\"check-roles form-group\\"><label class=\\"form-label\\">Roles <a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://github.com/bcgov/PIMS/wiki/Architecture-Security\\">Role Descriptions</a></label>
          <div class=\\"required form-group\\"><span class=\\"required\\">*</span><select multiple=\\"\\" name=\\"roles\\" required=\\"\\" data-testid=\\"role\\" id=\\"input-roles\\" class=\\"form-select form-control\\">
              <option value=\\"1\\" class=\\"option\\">roleVal</option>
            </select></div>
        </div>
        <p class=\\"label undefined\\">Notes</p>
        <div class=\\"form-group\\"><textarea name=\\"note\\" data-testid=\\"note\\" placeholder=\\"A note about this user\\" type=\\"text\\" id=\\"input-note\\" class=\\"form-control\\">test note</textarea></div>
        <div class=\\"is-disabled form-group\\"><label class=\\"form-label\\">Disable Account?&nbsp;</label><input name=\\"isDisabled\\" data-testid=\\"isDisabled\\" type=\\"checkbox\\" value=\\"false\\"></div>
        <div class=\\"justify-content-md-center row\\">
          <div role=\\"toolbar\\" class=\\"cancelSave btn-toolbar\\"><button type=\\"button\\" class=\\"mr-5 btn btn-secondary\\">Cancel</button><button type=\\"submit\\" class=\\"mr-5 btn btn-primary\\">Save</button></div>
        </div>
      </form>
    </div>
  </div>
</div>,"
`;
